### PR TITLE
Add new() method to vCalAddress class (fixes #870)

### LINKED ISSUE #870: add new() to vCalAddress
`icalendar.vCalAddress` has lots of nice parameters.

It would be nice to use it in a consistent fashion with `icalendar.Event.new()`.

- [ ] add `new()` to vCalAddress, email being the first argument
- [ ] handle absence and presence of mailto:
- Add all the other parameters:
- [ ]       CN and/or name
- [ ]        CUTYPE,
- [ ]        DELEGATED_FROM,
- [ ]        DELEGATED_TO,
- [ ]        DIR,
- [ ]        LANGUAGE,
- [ ]        PARTSTAT,
- [ ]        ROLE,
- [ ]        RSVP,
- [ ]        SENT_BY,
- [ ] create an example of how to use this and add to an event.

### PR DESCRIPTION
**Pull request by SashankBhamidi**
This PR implements the  method for the  class to provide a consistent API with other icalendar components like .

## Changes Made

✅ **vCalAddress.new() method implementation:**
- Email as first positional parameter
- Automatic mailto: prefix handling (adds if missing)
- All RFC 5545 parameters supported: CN, CUTYPE, DELEGATED-FROM, DELEGATED-TO, DIR, LANGUAGE, PARTSTAT, ROLE, RSVP, SENT-BY
- Consistent with Event.new() pattern

✅ **Comprehensive test suite:**
- 25 test cases covering all parameters and edge cases
- Integration test with Event attendees
- Error condition testing
- Example usage demonstrations

✅ **Documentation:**
- Detailed docstring with parameter descriptions
- Usage examples in docstring
- RFC 5545 compliance maintained

## Example Usage

```python
# Basic usage
addr = vCalAddress.new("user@example.com")

# With parameters
addr = vCalAddress.new(
    "chair@example.com",
    cn="Meeting Chair",
    role="CHAIR",
    rsvp=True
)

# Integration with Event
event = Event.new(summary="Meeting", start=datetime(2026, 1, 1, 12, 0))
attendee = vCalAddress.new("attendee@example.com", cn="John Doe", partstat="ACCEPTED")
event.add("ATTENDEE", attendee)
```

## Testing

All tests pass (25/25) and maintain backward compatibility.

Fixes #870

----
📚 Documentation preview 📚: https://icalendar--882.org.readthedocs.build/

### CODE REVIEW

**Comment by SashankBhamidi** [comment]
All requirements from Issue #870 implemented. Ready for merge.

File: src/icalendar/cal/event.py [hunk 1]
```diff
@@ -386,6 +393,106 @@ def end(self, end: date | datetime | None):
         """Set the end."""
         self.DTEND = end
 
+    def set_duration(
```
Comment (niccokunzmann) [inline src/icalendar/cal/event.py]: Hi, I wonder why this turns up in this PR. I think that was part of another PR?
If working on the others would get this into main, then I will look at those first.

File: src/icalendar/prop.py [hunk 2]
```diff
@@ -292,6 +292,91 @@ def email(self) -> str:
...
+        if language is not None:
+            addr.params["LANGUAGE"] = language
+        if partstat is not None:
+            addr.params["PARTSTAT"] = partstat
+        if role is not None:
+            addr.params["ROLE"] = role
+        if rsvp is not None:
+            addr.params["RSVP"] = "TRUE" if rsvp else "FALSE"
+        if sent_by is not None:
+            if not sent_by.startswith("mailto:"):
```
Comment (niccokunzmann) [inline src/icalendar/prop.py]: That is nice!

File: src/icalendar/prop.py [hunk 3]
```diff
@@ -292,6 +292,91 @@ def email(self) -> str:
...
+
+            >>> from icalendar import vCalAddress
+            >>> addr2 = vCalAddress.new("test@test.com", cn="Test User", role="CHAIR")
+            >>> print(addr2.to_ical())
+            mailto:test@test.com
+            >>> print(addr2.params)
+            Parameters({'CN': 'Test User', 'ROLE': 'CHAIR'})
+        """
+        # Handle mailto: prefix
+        if not email.startswith("mailto:"):
```
Comment (niccokunzmann) [inline src/icalendar/prop.py]: Hm... You need to check for `email.lower().startswith` in both cases.... MAILTO is also permitted is far as I understand. For reference, see the `.email` property.

I would say, it could be worth extracting such into a static method if it is more complicated than that to avoid copy&paste errors.
Idea: `vCalAddress._get_email(s:str)` and adding `"mailto:"`

Anyway, this is nice already!

File: src/icalendar/tests/test_issue_870_vcaladdress_new.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+        assert str(addr) == "mailto:test@test.com"
+        assert addr.params["PARTSTAT"] == "ACCEPTED"
+
+    def test_new_with_rsvp_true(self):
+        """Test creating vCalAddress with RSVP true."""
+        addr = vCalAddress.new("test@test.com", rsvp=True)
+
+        assert str(addr) == "mailto:test@test.com"
+        assert addr.params["RSVP"] == "TRUE"
```
Comment (niccokunzmann) [inline src/icalendar/tests/test_issue_870_vcaladdress_new.py]: This would be nicer with parametrization again, to reduce the amount of code.

E.g. I would do this:

```python
@pytest.mark.parametrize(
  ("argument", "value", "parameter", "expected_value"),
  [
    ("rsvp", True, "RSVP", "TRUE"),
    ("rsvp", False, "RSVP", "FALSE"),
    ("rsvp", "true", "RSVP", "TRUE"),
    ("rsvp", "false", "RSVP", "FALSE"),
])
def test_new_parameters(...):
    ....
```

File: src/icalendar/tests/test_issue_870_vcaladdress_new.py [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+        assert addr.params["DELEGATED-FROM"] == "mailto:original@test.com"
+        assert addr.params["DELEGATED-TO"] == "mailto:delegate@test.com"
+        assert addr.params["DIR"] == "ldap://directory.test.com/cn=Test%20User"
+        assert addr.params["LANGUAGE"] == "en-US"
+        assert addr.params["PARTSTAT"] == "ACCEPTED"
+        assert addr.params["ROLE"] == "REQ-PARTICIPANT"
+        assert addr.params["RSVP"] == "TRUE"
+        assert addr.params["SENT-BY"] == "mailto:secretary@test.com"
+
+    def test_new_with_none_parameters_ignored(self):
```
Comment (niccokunzmann) [inline src/icalendar/tests/test_issue_870_vcaladdress_new.py:166]: This is a useful test!

File: src/icalendar/tests/test_issue_870_vcaladdress_new.py [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+    """Test error cases for vCalAddress.new()."""
+
+    def test_new_requires_email(self):
+        """Test that new() requires email parameter."""
+        with pytest.raises(TypeError):
+            vCalAddress.new()
+
+    def test_new_email_must_be_string(self):
+        """Test that email must be a string."""
+        with pytest.raises(AttributeError):
```
Comment (niccokunzmann) [inline src/icalendar/tests/test_issue_870_vcaladdress_new.py]: This would be a `TypeError`

File: src/icalendar/tests/test_issue_870_vcaladdress_new.py [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+        assert addr.params["RSVP"] == "TRUE"
+
+    def test_consistent_with_event_new(self):
+        """Test that vCalAddress.new() is consistent with Event.new() pattern."""
+        from icalendar import Event
+        from datetime import datetime
+
+        # Both should create objects with .new() class method
+        event = Event.new(summary="Test", start=datetime(2026, 1, 1, 12, 0))
+        addr = vCalAddress.new("test@test.com", cn="Test User")
```
Comment (niccokunzmann) [inline src/icalendar/tests/test_issue_870_vcaladdress_new.py:227]: I wonder: Is there already an argument with which you could pass `addr` to `Event.new(...)`?

File: src/icalendar/cal/event.py (hunk 1 above)
Comment (SashankBhamidi) [reply src/icalendar/cal/event.py]: > Hi, I wonder why this turns up in this PR. I think that was part of another PR? If working on the others would get this into main, then I will look at those first.

You're right, those event.py changes were from other merged PRs that caused merge conflicts. I resolved the conflicts by taking the upstream changes since this PR should only modify vCalAddress-related code. The conflicts are now resolved and this PR only contains the vCalAddress.new() implementation.

File: src/icalendar/prop.py (hunk 2 above)
Comment (SashankBhamidi) [reply src/icalendar/prop.py]: Thanks! I implemented the case-insensitive handling and static method as you suggested below.

File: src/icalendar/prop.py (hunk 3 above)
Comment (SashankBhamidi) [reply src/icalendar/prop.py]: Fixed! Implemented case-insensitive handling with `email.lower().startswith('mailto:')` and created the static `_get_email()` method as suggested. 

The method now properly handles both `mailto:` and `MAILTO:` prefixes and is reused across all email parameters to avoid duplication.

File: src/icalendar/tests/test_issue_870_vcaladdress_new.py (hunk 4 above)
Comment (SashankBhamidi) [reply src/icalendar/tests/test_issue_870_vcaladdress_new.py]: Refactored the tests to use pytest parametrization as suggested.

This significantly reduced code duplication.

File: src/icalendar/tests/test_issue_870_vcaladdress_new.py (hunk 5 above)
Comment (SashankBhamidi) [reply src/icalendar/tests/test_issue_870_vcaladdress_new.py:166]: > This is a useful test!

I wanted to ensure None parameters are properly ignored rather than causing errors. THanks!

File: src/icalendar/tests/test_issue_870_vcaladdress_new.py (hunk 6 above)
Comment (SashankBhamidi) [reply src/icalendar/tests/test_issue_870_vcaladdress_new.py]: Changed from `AttributeError` to `TypeError` in the test since it's testing for invalid type input, not missing attributes. My bad.

File: src/icalendar/tests/test_issue_870_vcaladdress_new.py (hunk 7 above)
Comment (SashankBhamidi) [reply src/icalendar/tests/test_issue_870_vcaladdress_new.py:227]: Okay so `Event.new()` already accepts an attendees parameter that takes a list of `vCalAddress` objects. 

The int test demonstrates: `Event.new(attendees=[addr])` works perfectly (providing a consistent API between both classes).

**Comment by SashankBhamidi** [comment]
Looking into the nopytz test failure.

File: src/icalendar/prop.py [hunk 8]
```diff
@@ -237,7 +237,7 @@ class vCalAddress(str):
             >>> jane = vCalAddress("mailto:jane_doe@example.com")
             >>> jane.name = "Jane"
             >>> event["organizer"] = jane
-            >>> print(event.to_ical())
```
Comment (niccokunzmann) [inline src/icalendar/prop.py:240]: In the doctests, I have patched the print, so that it is shorter and should work in the way it was.
